README.md

```diff
@@ -7,3 +7,258 @@ instances for parsing and rendering. A convenient DSL to define those
+  = XmlPerson {_xpName :: !Text, _xpEmail :: !(Maybe Text)}
+      -- Defined at Tutorial.hs:13:1
+instance Show XmlPerson -- Defined at Tutorial.hs:13:1
+instance Eq XmlPerson -- Defined at Tutorial.hs:13:1
+instance FromDom XmlPerson -- Defined at Tutorial.hs:13:1
+instance ToXML XmlPerson -- Defined at Tutorial.hs:13:1
+
+```
+
+We have
+[FromDom](https://hackage.haskell.org/package/dom-parser-3.1.0/docs/Text-XML-DOM-Parser-FromDom.html#t:FromDom)
+and
+[ToXML](https://hackage.haskell.org/package/xml-conduit-writer-0.1.1.2/docs/Text-XML-Writer.html#t:ToXML)
+instance generated for us. That's because
+we instructed `xml-isogen` to generate them using the `Both` noun. You can specify
+also `Parser` or `Generator` if you want only one of them.
+
+The `_xpEmail` field is optional; that's because we prefixed it with `?` modifier.
+Here is the list of possible modifiers that affect types:
+
+Modifier | Description | Generated Type
+--- | --- | ---
+`!` | required | `a`
+`?` | optional | `Maybe a`
+`*` | repeated | `[a]`
+`+` | nonempty | `NonEmpty a`
+
+### Supported types
+
+Fields in a record may have any type as long as it's has instances for `Eq`, `Show`,
+`NFData`, `FromDom` (for parser) and `ToXml` (for generator). Remember though that
+TemplateHaskell requires types to be available before they are used in a splice.
+
+You can omit field type altogether, in that case type will be assumed to be a
+capitalized field name with `Xml` prefix.
+It's your responsibility to make sure that type exists.
+Example:
+
+```haskell
+newtype XmlEmail = XmlEmail Text
+  deriving (Eq, Show, NFData, ToXML)
+
+instance FromDom XmlEmail where
+  fromDom = XmlEmail <$> fromDom
+
+"Example1" =:= record Parser
+  ! "email"  -- will have type XmlEmail
+```
+
+### Enumerations
+
+Often XML element can contain only limited number of possible values. Lets define
+a type `Status` that can have only values `Active`, `Pending` or `Deleted`:
+
+```haskell
+"Status" =:= enum Both
+  & "Active"
+  & "Pending"
+  & "Deleted"
+```
+
+This definition will generate the following type for us:
+
+```haskell
+data XmlStatus
+  = XmlStatusActive
+  | XmlStatusPending
+  | XmlStatusDeleted
+  | UnknownXmlStatus !String
+```
+
+It has all the necessary instances, so you can use it as a type for a field.
+
+### Append content
+
+Sometimes XML you are dealing with contains a mix of elements and immediate content.
+Something like the following:
+
+```XML
+<?xml version="1.0" encoding="UTF-8"?>
+<example>
+    <field1>
+        I am
+    </field1>
+    totally
+    <field2>
+        weird
+    </field2>
+    LOL
+</example>
+```
+
+You can model this with an "append content" modifier `^`. It will instruct
+`xml-isogen` to append content of the field as it is, without wrapping it
+into an XML element. For our case it may look like this:
+
+```haskell
+"Example2" =:= record Both
+  ! "field1" [t|Text|]
+  ^ "mixed1" [t|Text|]
+  ! "field2" [t|Text|]
+  ^ "mixed2" [t|Text|]
+```
+
```

Comment:
You should probably show how the xml is would be parsed. (How would the `XmlExample2` record look after parsing the xml above)